### ISSUE TITLE
Compiler Bug: self-reference in extern method types

### ISSUE THREAD

**Issue by hackedy** [p4lang/p4c#2594]
Recursion in extern types causes a "Compiler Bug" error message.

Example: [selfref.p4.txt](https://github.com/p4lang/p4c/files/5426818/selfref.p4.txt)
```
extern E {
  void method(E e);
}
```

Output:
```
In file: /home/ryan/dev/p4c/frontends/p4/typeChecking/typeChecker.cpp:168
Compiler Bug: selfref.p4(1): Could not find type of <Type_Extern>(21) E/0 extern E<> {
  void method<>(E e); }
extern E {
       ^
```

**Comment by mbudiu-vmw** [comment]
I think this is an instance of https://github.com/p4lang/p4c/issues/2544.
The question is whether we want to support this idiom.

**Comment by hackedy** [comment]
Ah, yeah, looks like the same crash. This one may be more complicated to support because it introduces self-reference in types, which could break other parts of the typechecker if they expect types to not be circular.

**Comment by mbudiu-vmw** [comment]
Fixed by #2544
